lib/pkg.ml

```diff
@@ -233,48 +226,65 @@ let path_of_distrib p =
     | None -> "$(NAME)-$(TAG).tbz"
   in
   dev_repo_is_on_github p >>= fun a ->
-  homepage_is_on_github p >>| fun b ->
-  (if a || b then "releases/download/$(TAG)/" else "releases/") ^ basename
+  homepage_is_on_github p >>= fun b ->
+  let uri =
+    (if a || b then "releases/download/$(TAG)/" else "releases/") ^ basename
+  in
+  name p >>= fun name ->
+  tag p >>= fun tag ->
+  let defs = String.Map.(empty |> add "NAME" name |> add "TAG" tag) in
+  Pat.of_string uri >>| Pat.format defs
+
+let uri_of_dev_repo p =
+  opam_field_hd p "dev-repo"
+  >>| Option.map (fun dev_repo ->
```

Comment:
for 4.06 and 4.07 we must use `Stdext.Option`

Reply:
Thanks! I didn't know about that.